File: app/src/main/java/com/quickseries/quickseriesapp/AppComponent.java
package com.quickseries.quickseriesapp;

import com.quickseries.quickseriesapp.network.api.ApiClient;
import com.quickseries.quickseriesapp.network.wrapper.ApiWrapper;
import com.quickseries.quickseriesapp.network.wrapper.RealmWrapper;
import dagger.Component;
import org.jetbrains.annotations.NotNull;

import javax.inject.Singleton;

/**
 * Created by parth.gadhiya on 2018-10-26.
 */


@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(QuickSeriesApp target);

    void inject(@NotNull ApiClient apiClient);

    void inject(@NotNull RealmWrapper target);

    void inject(@NotNull ApiWrapper target);

}

File: app/src/main/java/com/android/kotlin/di/components/AppComponent.kt
package com.android.kotlin.di.components

import android.content.Context
import com.android.kotlin.di.ApplicationContext
import com.android.kotlin.di.modules.AppModule
import com.android.kotlin.di.modules.NetworkModule
import com.android.kotlin.utils.DataManager
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, NetworkModule::class))
interface AppComponent {

    @ApplicationContext
    fun context(): Context

    fun dataManager(): DataManager


}